core/os/input_event.cpp

```diff
@@ -34,8 +34,52 @@
  */
 
 bool InputEvent::operator==(const InputEvent &p_event) const {
+	if (type != p_event.type){
+		return false;
+	}
 
-	return true;
+	switch(type) {
```

Comment:
If you want to split the last commit you can do:

`git reset --soft HEAD^` (`--soft` to keep the changes)

Then commit the changes separately and force-push. I would recommend doing it in a cloned branch in case you screw up, and specify the branch like this:

`git push -f origin issue-6121-cloned:issue-6121`